### PR TITLE
patron_types: add a limit for unpaid subscription

### PR DESCRIPTION
**Pull request by zannkukai**
Add a circulation operation restriction if a patron has pending unpaid
subscriptions. If this limit is enabled, any output circulation
operation should be denied (checkout/renew/request).
Closes rero/rero-ils#1530.

Co-Authored-by: Renaud Michotte <renaud.michotte@gmail.com>

## Migration

An alembic script is available to migrate existing PatronTypes : `90d857fb5c23_unpaid_subscription_limit`

## Code review check list

- [ ] Commit message template compliance.
- [ ] Commit message without typos.
- [ ] File names.
- [ ] Functions names.
- [ ] Functions docstrings.
- [ ] Unnecessary commited files?
- [ ] Cypress tests successful?

### CODE REVIEW

File: rero_ils/modules/items/views/api_views.py [hunk 1]
```diff
@@ -379,10 +379,11 @@ def item(item_barcode):
...
             for action in item_dumps.get('actions', []):
                 if action == 'checkout':
-                    if item.number_of_requests() > 0:
-                        if item.patron_request_rank(patron) == 1:
-                            new_actions.append(action)
-                    else:
+                    if (
+                        item.number_of_requests() > 0
+                        and item.patron_request_rank(patron) == 1
+                        or item.number_of_requests() <= 0
```
Comment (jma) [inline rero_ils/modules/items/views/api_views.py:385]: This seems not related to the commit message: looks like to be a fix, no? If yes can you add an entry in the commit message?

File: tests/data/data.json [hunk 2]
```diff
@@ -1192,6 +1192,9 @@
     "description": "children patrons",
     "organisation": {
       "$ref": "https://bib.rero.ch/api/organisations/org1"
+    },
+    "limits": {
+      "unpaid_subscription": false
```
Comment (jma) [inline tests/data/data.json:1197]: Is it required? If yes this needs data migration steps.

**Review by jma — APPROVED** [review]
You can add in the commit message: several refactoring. Thanks for all.

File: tests/data/data.json (hunk 2 above)
Comment (zannkukai) [reply tests/data/data.json:1197]: @jma I include a alembic script to migrate existing PatronTypes ;-)

File: rero_ils/modules/items/views/api_views.py (hunk 1 above)
Comment (zannkukai) [reply rero_ils/modules/items/views/api_views.py:385]: OK

**Review by rerowep — APPROVED** [review]

**Review by lauren-d — APPROVED** [review]

**Comment by ManaDeweerdt** [comment]
- [ ] Editing the patron type, I can see that the limit is activated but it is displayed like this.
image

**Comment by zannkukai** [comment]
> * The option should be displayed closed to the subscription activation, and ideally should be displayed only if subscription are activated...

I can't move the options outside of the limits block ; but I placed it at first position and only visible if the `subscription_amount` is > 0
